spec/requests/admin/themes_controller_spec.rb

```diff
@@ -1129,6 +1129,65 @@
     end
   end
 
+  describe "#update_translations" do
+    fab!(:theme)
+
+    before do
+      theme.set_field(
+        target: :translations,
+        name: :en,
+        value: { en: { group: { hello: "Hello there!" } } }.deep_stringify_keys.to_yaml,
+      )
+      theme.save!
+    end
+
+    context "when logged in as an admin" do
+      before { sign_in(admin) }
+
+      it "should update a theme translation" do
+        put "/admin/themes/#{theme.id}",
+            params: {
+              theme: {
+                translations: {
+                  group: {
+                    hello: "Hello there! updated",
+                  },
+                },
+                locale: "en",
+              },
+            }
+
+        expect(response.status).to eq(200)
+        theme.reload
+
+        theme.translations.map { |t| expect(t.value).to eq("Hello there! updated") }
```

Comment:
I feel that I'm doing something wrong here.

The value in the theme does not get updated and the `response.parsed_body` also returns a HTML content instead of a json body

Maybe I'm doing something wrong in the params'?

Reply:
Yeah, it was two small issues I think: 

- the request needs `.json` appended
- the params were slightly incorrect

I pushed fixes for both, should be good now.

Reply:
Thanks a lot! I've added a few more tests as well, for checking when sending the wrong locale.